File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //Prompt for input
    long cardnum = -1;
    do
    {
        cardnum = get_long("Number: ");
    }
    while (cardnum < 0);

    //Calculate checksum
    int sum = 0;

    //Sum of every other digit starting with second to last
    for (long i = 100; i < cardnum * 10; i *= 100)
    {
        long rem = ((cardnum % i) / (i / 10)) * 2;
        if (rem >= 10)
        {
            sum += rem % 10;
            sum += rem / 10;
        }
        else
        {
            sum += rem;
        }
    }

    //Add remaining digits to sum
    for (long i = 10; i < cardnum * 10; i *= 100)
    {
        long rem = ((cardnum % i) / (i / 10));
        if (rem > 10)
        {
            sum += rem % 10;
            sum += rem / 10;
        }
        else
        {
            sum += rem;
        }
    }

    // Check card length and starting digits. Print AMEX, MASTERCARD, VISA or INVALID
    if ((sum % 10 == 0) && (sum > 0))
    {
        // Check if card is MASTERCARD
        if ((cardnum <= 5599999999999999) && (cardnum >= 5100000000000000))
        {
            printf("MASTERCARD\n");
        }
        // Otherwise check if card is AMEX
        else if ((cardnum >= 340000000000000 && cardnum <= 349999999999999) || (cardnum >= 370000000000000 && cardnum <= 379999999999999))
        {
            printf("AMEX\n");
        }
        // Otherwise check if card is VISA
        else if ((cardnum >= 4000000000000 && cardnum <= 4999999999999) || (cardnum >= 4000000000000000 && cardnum <= 4999999999999999))
        {
            printf("VISA\n");
        }
        // Otherwise card is INVALID
        else
        {
            printf("INVALID\n");
        }
    }
    // If sum == 0 and remainder != 0 => card is INVALID
    else
    {
        printf("INVALID\n");
    }
}

File: sentimental-credit/credit.py
# Import cs50 library
from cs50 import get_int


def main():
    # Execute prompt for user input
    cardnum = get_cardnum()
    # Cast cardnum to string for easy comparison
    cardstr = str(cardnum)
    # Calculate length of card number
    length = len(cardstr)

    # Initiate checksum
    sum = 0
    cardcpy = cardnum

    # Loop through every character
    for i in range(length):
        # For every second character add modulo of 10
        if i % 2 == 0:
            sum += cardcpy % 10
        # When character count is uneven add every digit of the double of the modulo of 10
        else:
            dig = cardcpy % 10 * 2
            sum += dig // 10 + dig % 10
        # Divide cardcpy by 10 and floor the value to an integer,
        # this makes the next iteration of the loop work with the next digit
        cardcpy //= 10

    # Check card length and starting digits. Print AMEX, MASTERCARD, VISA or INVALID
    if (sum % 10 == 0) and (sum > 0):
        # Check if card is MASTERCARD
        if (length == 16) and (cardstr[0:2] in ['51', '52', '53', '54', '55']):
            print("MASTERCARD")
        # Otherwise check if card is AMEX
        elif (length == 15) and (cardstr[0:2] in ['34', '37']):
            print("AMEX")
        # Otherwise check if card is VISA
        elif (length in [13, 16]) and (cardstr[0] == '4'):
            print("VISA")
        # Otherwise card is INVALID
        else:
            print("INVALID")
    # If sum == 0 and remainder != 0 => card is INVALID
    else:
        print("INVALID")


# Prompt for input
def get_cardnum():
    while True:
        cardnum = get_int("Number: ")
        if cardnum > 0:
            return cardnum


main()